### ISSUE TITLE
Please use the description of a referenced schema

### ISSUE THREAD

**Issue by TSheahan** [coveooss/json-schema-for-humans#32]
**Please use the description of a referenced item, if the referring item does not have a description set.**

Having a description on a ref adds vital readability to a rendered schema.

json-schema-for-humans already supports a description in a referring object, i.e.:
`name: { "$ref": "#/definitions/filled_string", description: "render this" }`

[Documentation](https://json-schema.org/understanding-json-schema/structuring.html#reuse) implies that description might be better inherited from the referenced object:

> You will always use $ref as the only key in an object: any other keys you put there will be ignored by the validator.

While it ought still to be okay to use a description on the referencing object if one is present, **it seems more desirable to detect and use the description from the referenced object, rather than to print no description**.

Considering the schema:
```
definitions:
  filled_string:
    description: a filled string
    type: string
    minLength: 1
type: object
additionalProperties: false
properties:
  name: { "$ref": "#/definitions/filled_string" }
  alignment: { "$ref": "#/definitions/filled_string" }
```

A more ideal rendering of this schema would put the description "a filled string" next to a ref link, rather than leaving the description blank.

This change could be done by adjusting the action of this part of the template:

https://github.com/coveooss/json-schema-for-humans/blob/f29cdab9f7b84e21dfd056b7a599f93e6b60652c/json_schema_for_humans/templates/schema_doc.template.html#L59-L60

**Comment by TSheahan** [comment]
Thanks for your attention, and fantastic work releasing the recursion fix, which also helped me out!

**Comment by dblanchette** [comment]
Hi @TSheahan, I actually consider it a bug, because the current behaviour of json-schema-for-humans is supposed to be to take the description next to $ref if available, fallback to the one in the referenced element.

I looked into it a bit, but it's not very simple to fix with the way it is currently structured. The `resolve_ref` filter does not return the referenced element, just the path to it.

I do have a solution in mind. If everything goes well, I should be able to provide a fix over the weekend.

Thanks for your interest and for the report!

**Comment by dblanchette** [comment]
Fix released in version 0.13.0
